### ISSUE TITLE
Q: User-Agent not taken from supplied request  (-r)?

### ISSUE THREAD

**Issue by drwetter** [sqlmapproject/sqlmap#3248]
To me it looks like it is not the case (logfile says ``sqlmap/1.2.8.15#dev`` and I supplied a proper requestfile including a user-agent already.). Is the additional ``--user-agent`` flag needed?

If so: Can't it just be retrieved from the request supplied if existing?

**Comment by stamparm** [comment]
My tests show differently (latest `1.2.9.36#dev`):

![ua1](https://user-images.githubusercontent.com/921555/45907353-d9982a00-bdf7-11e8-8b41-c3ca5f24b0db.png)

![ua2](https://user-images.githubusercontent.com/921555/45907357-dc931a80-bdf7-11e8-895c-9fca642d3b43.png)

**Comment by drwetter** [comment]
Don't know what happened. 

To explain: Due to a server side configuration problem I had access to a logfile. That was an application log, not a web server log. That had other on-related oddities in there as well as it turned out later. I suspect that during my tests there were tries on the user agent to exploit with the own sqlmap banner and that's what I must have seen.

I did a git pull and ran on other parts of the application sqlmap again with a higher verbosity and in fact I can't reproduce what spotted before. (The way I ran it always generates a script starting sqlmap in a reproducible fashion)

Thanks anyway for listening -- and the great tool!
